src/test/java/com/arpnetworking/metrics/impl/BaseFileSinkTest.java

```diff
@@ -21,7 +21,7 @@
 import ch.qos.logback.classic.spi.ILoggingEvent;
 import ch.qos.logback.core.encoder.Encoder;
 import ch.qos.logback.core.rolling.RollingFileAppender;
-import ch.qos.logback.core.rolling.TimeBasedRollingPolicy;
+import ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy;
```

Comment:
This version of logback does not correctly support the maximum number of files to rotate when size based rotation is enabled. You can either ugprade Logback and verify the desired behavior or we have a custom policy in logback-steno (which is already a dependency) which correctly supports size based rotation with random time based rotation as well:

https://github.com/ArpNetworking/logback-steno/blob/master/src/main/java/com/arpnetworking/logback/SizeAndRandomizedTimeBasedFNATP.java